rawkit/raw.py

```diff
@@ -175,6 +176,52 @@ def save_thumb(self, filename=None):
         self.libraw.libraw_dcraw_thumb_writer(
             self.data, filename.encode('ascii'))
 
+    def get_4_col_raw(self):
+        """
+        Read the 4 colour raw data
+        The returned numpy array is (given colour RGGB)
+        ---------
+        R B R B ... 
+        G B G B ...
+        R B R B ....
+        . . . . .
+        . . . .  .
+        . . . .   .
+        --------
+
+
+        Returns:
+            numpy array: 4 colour data of the image in unit16. (width x height)
+            str : colour channel description (ie. RGGB, RGBG)
+        """
+        # Unpack the data, so that rawdata is populated
+        self.unpack()
+        rawdata = self.data.contents.rawdata
+
+        # Raise error if 4 color image isn't there, which will happen for some
+        # cameras
+        raise_if_error(rawdata.color4_image.contents == ctypes.c_voidp)
```

Comment:
Yah, I just meant that a regular check (not this method) will work fine. This method actually has logic that's not valid for the comparison, it just happens to work because of the way Python type massages booleans.

```python
if rawdata.color4_image.contents is None:
    return None
```

will probably work (I'm not actually sure that we want this to be an error; since there just isn't any raw data, returning None makes sense to me). Then again, I'm not srue what would make a better API, maybe we need to make a `NoDataError` or something. @campaul?